app_dart/lib/src/service/branch_service.dart

```diff
@@ -30,6 +34,44 @@ class BranchService {
   final GerritService gerritService;
   final RetryOptions retryOptions;
 
+  /// Add a [CreateEvent] branch to Datastore.
+  Future<void> handleCreateRequest(CreateEvent createEvent) async {
+    log.info('the branch parsed from string request is ${createEvent.ref}');
+
+    final String? refType = createEvent.refType;
+    if (refType == 'tag') {
+      log.info('create branch event was rejected because it is a tag');
+      return;
+    }
+    final String? branch = createEvent.ref;
+    if (branch == null) {
+      log.warning('Branch is null, exiting early');
+      return;
+    }
+    final String repository = createEvent.repository!.slug().fullName;
```

Comment:
Cocoon's best practice is to keep all repository info as `RepositorySlug` and inline any adaptations.

So as below,

```dart
final String id = '${slug.fullName}/$branch'
```

Reply:
Umm maybe I am wrong, but I thought in the current state, the `id` is already
`${createEvent.repository!.slug().fullName}/$branch`
which aligns with the `${slug.fullName}/$branch` format?

Reply:
To clarify, this is about passing the slug directly and not fullName. Repository is ambiguous, which is why we use `RepositorySlug`